plugins/node/opentelemetry-instrumentation-hapi/src/utils.ts

```diff
@@ -119,3 +120,12 @@ export const getExtMetadata = (
     name: `ext - ${extPoint}`,
   };
 };
+
+export const getPluginFromInput = <T>(
+  pluginObj: HapiPluginObject<T>
+): Hapi.Plugin<T, void> =>
+  'plugin' in pluginObj
+    ? 'plugin' in pluginObj.plugin
+      ? pluginObj.plugin.plugin
+      : pluginObj.plugin
+    : pluginObj;
```

Comment:
I think I'd rather have this written out as nested if-else statements. On first glance, it's somewhat hard to see what's going on here.

Reply:
I took this from the original PR. I also think that should be nicer to read.